src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -0,0 +1,238 @@
+            if (first)
+                return suggested;
+            else
+                return MaxKind(suggested, previous);
+        }
+
+        private static Type MaxKind(Type a, Type b)
+        {
+            if (a == typeof(string) || b == typeof(string))
+                return typeof(string);
+            if (a == typeof(float) || b == typeof(float))
+                return typeof(float);
+            if (a == typeof(bool) || b == typeof(bool))
+                return typeof(bool);
+            return typeof(string);
+        }
+
+        /// <summary>
+        /// Reads a text file as a DataFrame.
+        /// Follows pandas API.
+        /// </summary>
+        /// <param name="filename">filename</param>
+        /// <param name="sep">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="names">column names (can be empty)</param>
+        /// <param name="dtypes">column types (can be empty)</param>
+        /// <param name="nrows">number of rows to read</param>
+        /// <param name="guess_rows">number of rows used to guess types</param>
+        /// <param name="encoding">text encoding</param>
+        /// <param name="index">add one column with the row index</param>
+        /// <returns>DataFrame</returns>
+        public static DataFrame ReadCsv(string filename,
+                                char sep = ',', bool header = true,
+                                string[] names = null, Type[] dtypes = null,
+                                int nrows = -1, int guess_rows = 10,
+                                Encoding encoding = null, bool index = false)
+        {
+            return ReadStream(() => new StreamReader(filename, encoding ?? Encoding.ASCII),
+                              sep: sep, header: header, names: names, dtypes: dtypes, nrows: nrows,
+                              guess_rows: guess_rows, index: index);
+        }
+
+        /// <summary>
+        /// Reads a text file as a DataFrame.
+        /// Follows pandas API.
+        /// </summary>
+        /// <param name="createStream">function which creates a stream</param>
+        /// <param name="sep">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="names">column names (can be empty)</param>
+        /// <param name="dtypes">column types (can be empty)</param>
+        /// <param name="nrows">number of rows to read</param>
+        /// <param name="guess_rows">number of rows used to guess types</param>
+        /// <param name="index">add one column with the row index</param>
+        /// <returns>DataFrame</returns>
+        public static DataFrame ReadStream(Func<StreamReader> createStream,
+                                char sep = ',', bool header = true,
+                                string[] names = null, Type[] dtypes = null,
+                                long nrows = -1, int guess_rows = 10, bool index = false)
```

Comment:
We should be using standard .NET parameter names. `nrows` and `guess_rows` do not follow the guidelines - 

[https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-parameters](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-parameters)
[https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/parameter-design](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/parameter-design)

Reply:
Even `nRows` and `dTypes` still don't follow the naming guidelines.

Typically this would be `rowCount` and `types`.

---
In reply to: [313009266](https://github.com/dotnet/corefxlab/pull/2710#discussion_r313009266) [](ancestors = 313009266)